fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java

```diff
@@ -524,7 +529,9 @@ private CompletableFuture<byte[]> readBlock(@Nonnull IndexInput requestingInput,
             ));
         } catch (ExecutionException e) {
             // This would happen when the cache.get() fails to execute the lambda (not when the block's future is joined)
-            throw new RecordCoreException(e.getCause());
+            final RecordCoreException recordCoreException = new RecordCoreException(e.getCause());
```

Comment:
Should this use `CompletionExceptionLogHelper.asCause()` ?

Reply:
Yeah, that looks good. It is not identical behavior (the suppressed exceptions get added to the cause rather than to the `RecordCoreException`) but I think this makes sense.
Done.